### DISCUSSION TITLE
Even for simple end effector approaching "straight-line motion," Curubo still plans very distant and laborious trajectories.

### DISCUSSION THREAD

**Discussion by pmj110119** [NVlabs/curobo#353] [Software Q&A]
If it’s not a bug, please use discussions: https://github.com/NVlabs/curobo/discussions

Please provide the below information in addition to your issue:

1. cuRobo installation mode (choose from [python, isaac sim, docker python, docker isaac sim]): isaac sim
2. python version: 3.10
3. Isaac Sim version (if using):  4.0

------
Issue Details

In many cases, the end effector of the robotic arm planned by Curubo does not move in a "straight line" but takes a longer route (like the video below). 

https://github.com/user-attachments/assets/4b12bb65-cbb2-4b1d-a345-24722ac63288

How can we make Curubo's plan follow the principle of "shortest ee pose movement" as much as possible?  Is it possible to achieve the "shortest ee pose movement" constraint directly through other general settings or parameters?

I know the official provides a "Constrained Planning" example, which can constrain the robotic arm to move along a certain axis, but it requires manual constraints and cannot be flexibly integrated into our existing robot project.

**Comment by wuisky** [comment]
I had the same issue. Currently my best practice is,
1. execute IK calculation, get multiple solutions(curobo can do this)
2. select one solution by a certain criterion( for example, select the one with shortest length from current configuration)
3. plan by plan_single_js API instead of plan_single API.
